hastory-server/src/Data/Hastory/Server/Utils.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Utils where
+
+import Crypto.Hash (Digest, SHA256(..), hashWith)
+import qualified Data.ByteString.Char8 as C
+import Data.Hastory.Types (Entry(..), ServerEntry(..), SyncRequest(..))
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import Network.HostName (HostName)
+
+hashEntry :: Entry -> HostName -> Digest SHA256
+hashEntry Entry {..} host = hashWith SHA256 (unifiedData :: C.ByteString)
+  where
+    unifiedData =
+      mconcat
+        [ T.encodeUtf8 entryText
+        , C.pack $ show entryWorkingDir
+        , C.pack $ show entryDateTime
+        , T.encodeUtf8 entryUser
+        , C.pack host
+        ]
+
+toServerEntry :: SyncRequest -> ServerEntry
+toServerEntry syncRequest = ServerEntry cmd workingDir dateTime user hostName contentHash
+  where
+    cmd = entryText entry
+    workingDir = entryWorkingDir entry
+    dateTime = entryDateTime entry
+    user = entryUser entry
+    hostName = syncRequestHostName syncRequest
+    contentHash = T.pack . show $ hashEntry entry (T.unpack hostName)
```

Comment:
There should be a way to convert a SHA256 to a ByteString here then.

Reply:
`contentHash` field is a value of type `Digest SHA256` - saving us from converting here. ;)